lib/music/note.rb

```diff
@@ -2,6 +2,7 @@ module Music
   class Note
     NOTES = ['C', 'C#/Db', 'D', 'D#/Eb', 'E', 'F', 'F#/Gb', 'G', 'G#/Ab', 'A', 'A#/Bb', 'B']
     NOTE_STRINGS = ['Ab', 'A', 'A#', 'Bb', 'B', 'C', 'C#', 'Db', 'D', 'D#', 'Eb', 'E', 'F', 'Gb', 'G', 'G#']
+    ENHARMONICS = {'E#' => 'F', 'Fb' => 'E', 'B#' => 'C', 'Cb' => 'B'}
```

Comment:
I'm not sure that I've ever heard of enharmonics before, so bear with me 😄 

I'm looking at [this page](http://musicnotation.org/tutorials/enharmonic-equivalents/) and [this piano key layout](http://www.piano-keyboard-guide.com/wp-content/uploads/2018/12/Piano-Keys-and-Notes-Piano-Keyboard-Diagram-1.jpg) as references and it seems like sharps and flats move you up and down respectively to the next note (including black keys).

So what it seems like you have here are the special-cases cases of enharmonics which aren't just moving up or down one letter.  From the code below where you check to see if something is in this hash and then fallback that seems to be the case.  Is that right?

Reply:
Yes that's correct, I'm butchering the term enharmonic in that sense. these are more like special case enharmonics. You would only spell the note as E# if you are playing in the key of C# (7 sharps) or F# (6 sharps), any other time you spell it as F.